### ISSUE TITLE
pauseOnInteraction doesn't exist on toast machine

### ISSUE THREAD

**Issue by RIP21** [chakra-ui/zag#1752]
# 🐛 Bug report

[Docs state about `pauseOnInteraction` functionality of a machine](https://zagjs.com/components/react/toast#pausing-the-toasts), but in reality it's non-existent and cannot be set. It's always `true` and the user cannot do anything about it.

## 💥 Steps to reproduce

Check your own code here :)
https://github.com/chakra-ui/zag/blob/main/packages/machines/toast/src/toast.types.ts#L150-L195

## 🧐 Expected behavior

It's possible to disable `pauseOnInteraction` and then implement your own interaction behavior. Now it requires 99999 hacks that are unreliable to force it to behave.

## 🧭 Possible Solution

Write some code to actually handle that :D

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Zag Version      |     0.65.0       |
| Browser          |        Chrome Latest    |
| Operating System |    MacOS Latest        |

**Comment by segunadebayo** [comment]
This needed a docs update instead. We don't intend to implement `pauseOnInteraction`, consider using the `api.pause(...)` if you really need this.

**Comment by RIP21** [comment]
But it's a not nice UX? Maybe.

I had to remove all the default event listeners and reimplement them instead of disabling this functionality via flag and then I used `api.pause` as I wanted.

```ts
    const mergedProps = mergeProps(api.getGroupProps({ placement }), rest)
    // here we remove event listeners that cannot be disabled programmatically
    delete mergedProps.onMouseMove
    delete mergedProps.onMouseLeave
    delete mergedProps.onBlur
 ```
 
And then other code to make it work the way we wanted, but I'm afraid I may miss some other default keyboard behavior because of that that Zag brings. So far so good, but just saying.

Is it a stupid approach, or a pretty smart hack? :)
